### PR TITLE
Redux in components

### PR DESCRIPTION
**Pull request by jadibdev**
### Description
Added Redux provider to connect the app to the store, added functionality to display books from the store, implemented add new book functionality, implemented remove book functionality.

### CODE REVIEW

File: src/components/BookInput.js [hunk 1]
```diff
@@ -1,11 +1,31 @@
...
+      >
+        <input type="text" placeholder="Title" required name="title" />
+        <input type="text" placeholder="Author" required name="author" />
+        <button
+          type="submit"
+        >
+          Add new book
+        </button>
+      </form>
+    </div>
```
Comment (julie-ify) [inline src/components/BookInput.js]: - [x] This component should have a local state with title and author, they should have their inputs that update the local state every time they update (onChange). Kindly create a local state with title and author and use the `onChange` event handler to update the inputs in the state.

File: src/components/Categories.js [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
+      >
+        <Link to="/">Books</Link>
+        {' '}
+        |
+        {' '}
+        <Link to="/categories">Categories</Link>
+      </nav>
+      <button type="button" onClick={() => { dispatch(checkStatus()); }}>Check Status</button>
+    </div>
+  );
```
Comment (julie-ify) [inline src/components/Categories.js]: - [x] When I click on the `Check status` button nothing happens, Kindly ensure that when a user clicks the `Check status` button, the page displays `Under construction` which you have in your categories reducer.

File: src/redux/categories/categories.js [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
+  id: 0,
+  title: 'Zero To One',
+  author: 'Peter Thiel',
+},
+{
+  id: 1,
+  title: 'Hit Refresh',
+  author: 'Satya Nadella',
+},
+];
```
Comment (julie-ify) [inline src/redux/categories/categories.js]: - [x] Kindly remove the objects you added in the initial state as this is not related to checking status. You can have just an empty array as the initial state. This will make your code more readable.

File: src/components/Books.js [hunk 4]
```diff
@@ -1,13 +1,30 @@
...
-  <div>
-    <Book title="The Pragmatic Programmer" author="by Andy Hunt and Dave Thomas" />
-    <button type="button">Remove</button>
-    <BookInput />
-  </div>
-);
+const Books = () => {
+  /* useSelector((state) => console.log('State: ', state.books[0].title));
+  const title = useSelector((state) => state.books[0].title);
+  const author = useSelector((state) => state.books[0].author); */
```
Comment (julie-ify) [inline src/components/Books.js]: - [ ] To have a cleaner and readable codebase, kindly remove these unnecessary commented lines of code.

File: src/redux/books/books.js [hunk 5]
```diff
@@ -1,8 +1,18 @@
...
+  title: 'Zero To One',
+  author: 'Peter Thiel',
+},
+{
+  id: 1,
+  title: 'Hit Refresh',
+  author: 'Satya Nadella',
+},
+];
+let currentId = 2;
```
Comment (julie-ify) [inline src/redux/books/books.js:15]: - [ ] **[OPTIONAL]** I suggest you make use of the `uuidv4` library to generate unique ids for your books. It is more authentic and less error-prone. Check this [link](https://www.codegrepper.com/code-examples/javascript/how+to+use+uuidv4+in+react) on the example code of how to use it in React.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @jadibdev,

Good job so far!

## Highlights!
- GitFlow was followed✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️  

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @jadibdev , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
